### PR TITLE
Upgrade ember

### PR DESCRIPTION
**Pull request by helloWorldCindy**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

Upgrade Ember-cli to 2.12.3

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "ember-pollboy",
   "dependencies": {
-    "ember": "^2.11.0",
+    "ember": "^2.12.2",
```
Comment (notmessenger) [inline bower.json]: Aren't we now using `npm`s `ember-source`?  This should't be needed.

File: package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
   "license": "MIT",
   "devDependencies": {
     "broccoli-asset-rev": "^2.5.0",
-    "ember-cli": "^2.11.0",
+    "ember-cli": "^2.12.3",
```
Comment (notmessenger) [inline package.json]: Remove the `^` - we're locking this dependency down.

File: package.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
   "license": "MIT",
   "devDependencies": {
     "broccoli-asset-rev": "^2.5.0",
```
Comment (notmessenger) [inline package.json]: Change to `^2.4.5` to be consistent with what `ember-cli` is currently at and with the other repos doing the same.

File: package.json [hunk 4]
```diff
@@ -22,7 +22,7 @@
...
-    "ember-cli": "^2.11.0",
+    "ember-cli": "^2.12.3",
     "ember-cli-app-version": "^2.0.1",
```
Comment (notmessenger) [inline package.json]: Remove this dependency

File: tests/index.html [hunk 5]
```diff
@@ -25,7 +25,7 @@
     <script src="{{rootURL}}assets/vendor.js"></script>
     <script src="{{rootURL}}assets/test-support.js"></script>
     <script src="{{rootURL}}assets/dummy.js"></script>
-    <script src="{{rootURL}}testem.js" integrity=""></script>
+    <script src="/testem.js" integrity=""></script>
```
Comment (notmessenger) [inline tests/index.html]: this should be the first `<script>` tag in this group

File: tests/helpers/start-app.js [hunk 6]
```diff
@@ -5,16 +5,13 @@ import Application from '../../app'
 import config from '../../config/environment'
```
Comment (notmessenger) [inline tests/helpers/start-app.js:5]: The entire contents of this file should be:

```
import Ember from 'ember'
const {merge, run} = Ember

import Application from '../../app'
import config from '../../config/environment'

export default function startApp (attrs) {
  let attributes = merge({}, config.APP)
  attributes = merge(attributes, attrs) // use defaults, but you can override;

  return run(() => {
    let application = Application.create(attributes)
    application.setupForTesting()
    application.injectTestHelpers()
    return application
  })
}
```

**Review by notmessenger — CHANGES_REQUESTED** [review]

**Comment by notmessenger** [comment]
These modifications need to be made to the _package.json_ file:

**change these packages to these values**
"ember-cli-htmlbars-inline-precompile": "^0.3.6"
"ember-cli-inject-live-reload": "^1.4.1"
"ember-export-application-global": "^1.0.5"
"ember-load-initializers": "^0.6.0"

**add these packages**
"ember-cli-shims": "^1.0.2"
"ember-source": "~2.12.0"

**remove these packages**
"ember-disable-proxy-controllers": "^1.0.1"

File: bower.json (hunk 1 above)
Comment (helloWorldCindy) [reply bower.json]: Fixed.

File: package.json (hunk 3 above)
Comment (helloWorldCindy) [reply package.json]: Fixed.

File: package.json (hunk 2 above)
Comment (helloWorldCindy) [reply package.json]: Fixed.

File: package.json (hunk 4 above)
Comment (helloWorldCindy) [reply package.json]: Fixed.

File: tests/index.html (hunk 5 above)
Comment (helloWorldCindy) [reply tests/index.html]: Fixed.

File: tests/helpers/start-app.js (hunk 6 above)
Comment (helloWorldCindy) [reply tests/helpers/start-app.js:5]: aren't they the same?

Comment (notmessenger) [reply tests/helpers/start-app.js:5]: Oops, sorry, yes they are.  Not sure what I was looking at.

**Comment by notmessenger** [comment]
Url in _.gitignore_ should be

```
-# See http://help.github.com/ignore-files/ for more about ignoring files.
+# See https://help.github.com/ignore-files/ for more about ignoring files.
```

File: .eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  extends: ['frost-standard'],
```
Comment (juwara0) [inline .eslintrc.js]: This line should not have the array brackets but instead be:

`extends: 'frost-standard',`

Comment (helloWorldCindy) [reply .eslintrc.js]: Fixed.

File: .travis.yml [hunk 8]
```diff
@@ -22,31 +22,17 @@ cache:
 
 env:
   matrix:
-  - EMBER_TRY_SCENARIO=ember-1-12
```
Comment (dafortin) [inline .travis.yml:25]: @juwara0 do we need `firefox: 'latest-esr'`

File: .travis.yml [hunk 9]
```diff
@@ -22,31 +22,17 @@ cache:
...
+  - EMBER_TRY_SCENARIO=ember-default
   global:
     - CXX=g++-4.8
     - secure: Ws3qXGyj8zHDQ83r4XxTq9bZGoY4r0ypuzZ2W/lHGjwQpkEwut79z5MTZd8BlYS/UWxruG04FyiftJmO9Sw9q3G/VAyxbXiLfeRF34bmh0Ymr4iNz/TYiRjw69S/0kuT1W0zV5sxB3JiE37O3T1bHyqZ/H9kMSUA0NGuGkZxoovM8bDCBBnYOnvOQAW70YEWccQO3w/mUmRX+OhUMblsR/y3uIBhmUn7wdCqOdT9PFZM68DWma6DhgcgRbtGjShJ0+jogeEVbb6mqeaPL5l/0sgBHSNY3SEOOKC6Hqo5FyhTvPayzqBtB5xDPehIgHmrJibs925Y6zlIWJHVFu4JjVNGOyf8GKEVN8yFxT5Nyma2hV5JAErCYIKttKiN298aJdkLw/VaHEdkgi2f+RY7Mrjj7eqD7nsdZ6LdMbVqqlDfzX7pKTktJwVxUkQy8WmdufG/LHgK4C6RVdcG7hPkr3IytCpsUMNIEnpOfBbrXG56ENK4akF39AmbagQWr4hExIWsKaYihH9VNnShunsBoAuSuLZaZPQi/VPvmM7J5U0Acaql33A8Rxy/iZ5x2SvqCYO/t2mKj5x72yYhUO/3YY/H5FcughRYUn0rQYvdiCU55JXJBbm0iGEWbRT8y38gaXwai058SeHhZPMe6lxsG6OZMXFuWRqZCFLF7+iNCrs=
 
 matrix:
   fast_finish: true
   allow_failures:
-  - env: EMBER_TRY_SCENARIO=ember-canary
+  - env: EMBER_TRY_SCENARIO=ember-release
```
Comment (dafortin) [inline .travis.yml]: we don't need `release` if it's not in the run matrix

File: bower.json [hunk 10]
```diff
@@ -1,8 +1,6 @@
 {
   "name": "ember-pollboy",
   "dependencies": {
-    "ember": "^2.11.0",
-    "ember-cli-shims": "0.1.0",
     "ember-mocha-adapter": "~0.3.1"
```
Comment (dafortin) [inline bower.json]: No necessary.

File: tests/.eslintrc [hunk 11]
```diff
@@ -1 +0,0 @@
-{}
```
Comment (dafortin) [inline tests/.eslintrc:1]: remove this file

Comment (helloWorldCindy) [reply tests/.eslintrc:1]: @dafortin Do you know why this file will show up here? I can't find this file in my local copy.

File: bower.json (hunk 10 above)
Comment (helloWorldCindy) [reply bower.json]: Fixed.

File: .travis.yml (hunk 9 above)
Comment (helloWorldCindy) [reply .travis.yml]: Fixed.

File: tests/.eslintrc (hunk 11 above)
Comment (dafortin) [reply tests/.eslintrc:1]: I'm not sure.

File: package.json [hunk 12]
```diff
@@ -21,34 +21,34 @@
...
     "ember-cli-code-coverage": "0.3.11",
     "ember-cli-dependency-checker": "^1.3.0",
     "ember-cli-htmlbars": "^1.1.1",
-    "ember-cli-htmlbars-inline-precompile": "0.3.6",
-    "ember-cli-inject-live-reload": "^1.6.1",
+    "ember-cli-htmlbars-inline-precompile": "^0.3.6",
+    "ember-cli-inject-live-reload": "^1.4.1",
     "ember-cli-mocha": "0.13.2",
+    "ember-cli-shims": "^1.0.2",
     "ember-cli-test-loader": "^1.1.1",
```
Comment (juwara0) [inline package.json]: This should be removed.

File: package.json [hunk 13]
```diff
@@ -21,34 +21,34 @@
...
     "ember-cli-uglify": "^1.2.0",
     "ember-disable-prototype-extensions": "^1.1.0",
-    "ember-disable-proxy-controllers": "^1.0.1",
-    "ember-export-application-global": "^1.1.1",
-    "ember-load-initializers": "0.6.3",
+    "ember-export-application-global": "^1.0.5",
+    "ember-load-initializers": "^0.6.0",
     "ember-resolver": "^2.1.1",
```
Comment (juwara0) [inline package.json]: This should be: `"ember-resolver": "^2.0.3",`

File: tests/.eslintrc.js [hunk 14]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  env: {
+    embertest: true
+  }
+};
```
Comment (juwara0) [inline tests/.eslintrc.js]: Please remove the semicolon

File: .travis.yml (hunk 8 above)
Comment (juwara0) [reply .travis.yml:25]: No, it can be removed. I have this listed as a task in the dev ops project to do across all the repos.

Comment (juwara0) [reply .travis.yml:25]: Fixed

File: package.json (hunk 12 above)
Comment (juwara0) [reply package.json]: Fixed

File: package.json (hunk 13 above)
Comment (juwara0) [reply package.json]: FIXED

File: tests/.eslintrc.js (hunk 14 above)
Comment (juwara0) [reply tests/.eslintrc.js]: FIXED

**Review by notmessenger — APPROVED** [review]

**Comment by notmessenger** [comment]
👍

**Comment by juwara0** [comment]
Approved.

**Pull request merged**